### PR TITLE
Add `caseSensitive` option

### LINKED ISSUE #9: Case sensitive option
hi all,

I saw it was discussed previously, but please consider adding case-sensitive option. I'm working on a project where I need to filter out an array of file names and use wildcards. Case does matter...

Code-wise, it's just a toggle on the regex flag and this feature would not break the existing apps, just implement default on "case insensitive = on" mode and let people turn it off.

What do you think?

Thanks for creating and maintaining this by the way.

### PR DESCRIPTION
**Pull request by Ricardo-Silva91**
fixes #9 

added global variable options, right now only holds the caseSensitive option but can be used for other features in the future.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,7 +3,16 @@ const escapeStringRegexp = require('escape-string-regexp');
 
 const reCache = new Map();
 
-function makeRe(pattern, shouldNegate) {
+global.options = {
```
Comment (SamVerschueren) [inline index.js]: Remove

File: index.js [hunk 2]
```diff
@@ -3,7 +3,16 @@ const escapeStringRegexp = require('escape-string-regexp');
...
+	caseSensitive: false
+};
+
+function makeRe(pattern, shouldNegate, options) {
+	if (options !== undefined) {
```
Comment (SamVerschueren) [inline index.js]:
```js
const opts = Object.assign({}, options, {
    caseSensitive: false
});
```

**Comment by Ricardo-Silva91** [comment]
hello again,  made the changes you requested, let me know if there's something else to edit.

**Comment by revelt** [comment]
Good stuff! I'd like to add my twopence; should we add some unit tests for hardcoded default `caseInsensitive = false`? Basically, to test the default setting with no opts, hardcoded default and then with opts on - three cases...

File: index.js (hunk 1 above)
Comment (revelt) [reply index.js]: @SamVerschueren I never saw `global` before so did some investigation and have a question for you. Node [documentation](https://nodejs.org/api/globals.html#globals_global) says that 

> ... In Node.js ... The top-level scope is not the global scope; var something inside a Node.js module will be local to that module.

If we use `global.`, would that mean that there's potential clash between `options`, let's say if couple other modules were using same `options` and same `caseSensitive` key? If `global.options` really sets the object to the global scope, then is it better if we kept options within the top level of the **module's** scope?

**Comment by Ricardo-Silva91** [comment]
Bump.
I understand this surely is not an important or urgent PR, but could you please spare a couple minutes just so we can close it.
Sorry to be such a bother.

File: index.js (hunk 2 above)
Comment (revelt) [reply index.js]: @SamVerschueren  The options and defaults above are in the wrong order above, but @Ricardo-Silva91 put it the correct way. Please review and approve.

**Comment by sindresorhus** [comment]
@Ricardo-Silva91 Thanks for your patience. Can you also document the option? Example on how to format it: https://github.com/sindresorhus/boxen#boxeninput-options

**Comment by Ricardo-Silva91** [comment]
No problem :smiley:

**Comment by revelt** [comment]
Guys, I looked at the last readme change and it felt it could be improved. What if we changed the readme chunk in question to the following:

> Use `*` to match zero or more characters. A pattern starting with `!` will be negated. Default mode is case-insensitive, but you can set it to case-sensitive using option `caseSensitive`, see above.
>
> In general, matching strings that mean _file system things_ (files, folders, etc) is best done in a case-insensitive mode. Matching strings that mean _internal values_, for example, JSON keys, should be done in case-sensitive mode.

When you think about it, it's elementary and basic stuff, but people (like I did) might just jump in _gung-ho_ and forget to set case-sensitive setting when dealing with case-sensitive values in object keys and so on.

What do you think?

**Comment by sindresorhus** [comment]
@revelt I think it's too much hand-holding. This project is for developers, not end-users. But I've added a short note about it.

**Comment by sindresorhus** [comment]
Nice work @Ricardo-Silva91 :)

**Pull request merged**